### ISSUE TITLE
Adding a long Lua Record using the API causes the record to be broken into chunks leading to potential failures

### ISSUE THREAD

**Issue by neilcook** [PowerDNS/pdns#14002]
- [ ] This is not a support question, I have read [about opensource](https://www.powerdns.com/opensource.html) and will send support questions to the IRC channel, [Github Discussions](https://github.com/PowerDNS/pdns/discussions/) or the mailing list.
- [x ] I have read and understood the ['out in the open' support policy](https://blog.powerdns.com/2016/01/18/open-source-support-out-in-the-open/)

 - Program: Authoritative,  
 - Issue type: Bug report

### Short description

When adding a LUA record using the Auth API, the Auth server breaks the record up into chunks of 259 characters, like \"xxx\" \"yyy\" without first checking whether it is doing so on "whitespace" characters. This can break the Lua record. 

### Environment

 - Operating system: Any
 - Software version: Auth master
 - Software source: Tested using powerdns/pdns-auth-master:latest image 

### Steps to reproduce

1. Create a Lua record of 260 bytes or longer. For example:
```
A \";answers = {{'10.20.58.58', true}, {'10.178.60.249', true}, {'10.183.96.244', true}, {'10.20.61.58', true}} function pickrandomup(t) local lt = {} for i, v in ipairs(t) do if v[2] then table.insert(lt, i, v[1]) end end return pickrandom(lt) end return pickrandomup(answers)\"
```
2. Add the record to the Auth server using the REST API
3. Check how the record is stored in the Auth server by retrieving the record using the REST API. The above record for example will be stored as:
```
A \";answers = {{'10.20.58.58', true}, {'10.178.60.249', true}, {'10.183.96.244', true}, {'10.20.61.58', true}} function pickrandomup(t) local lt = {} for i, v in ipairs(t) do if v[2] then table.insert(lt, i, v[1]) end end return pickrandom(lt) end return pic\" \"krandomup(answers)\"
4. Test the record using dig - it will fail because the above is not a valid LUA record.

### Expected behaviour
I would expect the Auth server to behave one of two ways:
1) Either not break the LUA record into chunks
or
2) Break the Lua record into chunks, but only on existing whitespace characters (if any)
<!-- What would you expect to happen when the reproduction steps are run -->

### Actual behaviour
See above for the output from the REST API. Dig returns SERVFAIL. 
<!-- What did happen? Please (if possible) provide logs, output from `dig` and/or tcpdump/wireshark data -->

### Other information
<!-- if you already did more digging into the issue, please provide all the information you gathered -->

**Comment by Habbie** [comment]
This only happens with the LMDB backend, or when there is AXFR between the originating database and the instance that executes the code.

A way simpler fix would be for us to not insert a space when joining the chunks together, but that's a backwards incompatible change (although I do not know how many users would be affected). Perhaps it makes sense to make that change, and do it soon, so people will notice in 5.0-alpha1.

**Comment by Habbie** [comment]
retitled and reopened. 4.9 gets the fix from #14021, and for the future we'll see about making the API experience better.
